docs/source/python/timestamps.rst

```diff
@@ -0,0 +1,198 @@
+"Time Zone Naive".  Python/Pandas timestamp types with an associated time zone are
+referred to as "Time Zone Aware".
+
+
+Timestamp Conversions
+=====================
+
+Pandas/Arrow ⇄ Spark
+--------------------
+
+Spark stores timestamps as 64-bit integers representing microseconds since
+the UNIX epoch.  It does not store any metadata about time zones with its
+timestamps.
+
+Spark interprets timestamps with the *session local time zone*, (i.e.
+``spark.sql.session.timeZone``). If that time zone is undefined, Spark turns to
+the default system time zone. For simplicity's sake below, the session
+local time zone is always defined.
+
+This implies a few things when round-tripping timestamps:
+
+#.  Timezone information is lost (all timestamps that result from
+    converting from spark to arrow/pandas are "time zone naive").
+#.  Timestamps are truncated to microseconds.
+#.  The session time zone might have unintuitive impacts on 
+    translation of timestamp values. 
+
+Spark to Pandas (through Apache Arrow)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The following cases assume the Spark configuration
+``spark.sql.execution.arrow.enabled`` is set to ``"true"``.
+
+::
+
+    >>> pdf = pd.DataFrame({'naive': [datetime(2019, 1, 1, 0)], 
+    ...                     'aware': [Timestamp(year=2019, month=1, day=1, 
+    ...                               nanosecond=500, tz=timezone(timedelta(hours=-8)))]})
+    >>> pdf
+           naive                               aware
+           0 2018-10-01 2018-10-01 00:00:00.000000500-08:00
+
+    >>> spark.conf.set("spark.sql.session.timeZone", "UTC")
+    >>> utc_df = sqlContext.createDataFrame(pdf)
+    >>> utf_df.show()
+    +-------------------+-------------------+
+    |              naive|              aware|
+    +-------------------+-------------------+
+    |2019-01-01 00:00:00|2019-01-01 08:00:00|
+    +-------------------+-------------------+
+                    
+Note that conversion of the aware timestamp is shifted to reflect the time
+assuming UTC (it represents the same instant in time).  For naive
+timestamps, Spark treats them as being in the system local
+time zone and converts them UTC. Recall that internally, the schema
+for spark dataframe's does not store any time zone information with
+timestamps.
+
+Now if the session time zone is set to US Pacific Time (PST) we don't
+see any shift in the display of the aware time zone (it
+still represents the same instant in time):
+
+::
+
+    >>> spark.conf.set("spark.sql.session.timeZone", "US/Pacific")
+    >>> pst_df = sqlContext.createDataFrame(pdf)
+    >>> pst_df.show()
+    +-------------------+-------------------+
+    |              naive|              aware|
+    +-------------------+-------------------+
+    |2019-01-01 00:00:00|2019-01-01 00:00:00|
+    +-------------------+-------------------+
+
+Looking again at utc_df.show() we see one of the impacts of the session time
```

Comment:
I guess the purpose of this section is to show the weird behavior when session time zone is changed in the middle of a Spark session. My feeling is that this is not something that people should be doing and trying to understand that behavior is not too helpful to most user. 

I prefer to leave this out actually.

Reply:
That makes sense.  I think even though this is an uncommon case it gives a good mental model of how spark is representing/converting time zones.  I'd like to get a second opinion from someone else before removing it if that is OK.

Reply:
SGTM. This feels to me is more for "Advanced user" that really wants to understand the details. Have that be explicit would be helpful too if we keep this.